app/controllers/api/v1/recordings_controller.rb

```diff
@@ -78,6 +79,15 @@ def recording_params
         params.require(:recording).permit(:name)
       end
 
+      def recordings_sort_params
+        allowed_values = %w[ASC DESC]
+        columns = params.permit(sort: %i[name length visibility])[:sort].to_h
+        columns.transform_values do |v|
+          res = v if allowed_values.include?(v)
+          res || 'ASC'
+        end
+      end
```

Comment:
This is too complex and hard to read. I think it may be better if the API supports `sort: { column: 'name', direction: 'ASC' }` instead